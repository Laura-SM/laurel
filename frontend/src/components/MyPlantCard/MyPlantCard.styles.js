import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  cardContainer: {
    width: 175,
    height: 175,
    borderWidth: 2,
    borderColor: '#93B1A7',
    borderRadius: 10,
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10,
  },
  image: {
    width: 100,
    height: 100,
    borderRadius: 10,
    backgroundColor: '#E5E5E5',
  },
});

export default styles;
