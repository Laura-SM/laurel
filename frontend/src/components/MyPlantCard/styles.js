import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  cardContainer: {
    width: 165,
    height: 230,
    backgroundColor: '#93B1A7',
    borderRadius: 10,
    justifyContent: 'flex-end',
    alignItems: 'center',
    marginVertical: 10,
    marginLeft: 20,
  },
  image: {
    position: 'absolute',
    top: 0,
    width: 165,
    height: 165,
    borderRadius: 10,
    backgroundColor: '#E5E5E5',
  },
  text: {
    fontSize: 18,
    marginHorizontal: 20,
    marginBottom: 10,
  },
});

export default styles;
