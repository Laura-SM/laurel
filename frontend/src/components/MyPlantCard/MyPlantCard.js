import React from 'react';
import {Image, Pressable, Text} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import styles from './MyPlantCard.styles';
import globalStyles from '../../styles/global.styles';

const MyPlantCard = ({plant}) => {
  const navigation = useNavigation();
  return (
    <Pressable
      style={styles.cardContainer}
      onPress={() =>
        navigation.navigate('MyPlantDetails', {plantId: plant._id})
      }>
      <Image style={styles.image} source={{uri: plant.image}} />
      <Text style={globalStyles.boldText}>{plant.name}</Text>
      <Text style={globalStyles.text}>{plant.room}</Text>
    </Pressable>
  );
};

export default MyPlantCard;
