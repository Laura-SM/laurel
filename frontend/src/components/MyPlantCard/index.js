import React from 'react';
import {Image, Pressable, Text} from 'react-native';
import styles from './styles';
import globalStyles from '../../styles/globalStyles';

const MyPlantCard = ({plant, navigation}) => {
  return (
    <Pressable
      style={styles.cardContainer}
      onPress={() =>
        navigation.navigate('MyPlantDetails', {plantId: plant._id})
      }>
      <Image style={styles.image} source={{uri: plant.image}} />
      <Text style={globalStyles.boldText}>{plant.name}</Text>
      <Text style={globalStyles.text}>{plant.room}</Text>
    </Pressable>
  );
};

export default MyPlantCard;
