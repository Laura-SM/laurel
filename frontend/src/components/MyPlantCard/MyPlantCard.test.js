import React from 'react';
import MyPlantCard from './MyPlantCard';
import {render} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

describe('Given a PlantsSearcher component', () => {
  const mockStore = configureStore([thunk]);
  const store = mockStore({
    plants: [
      {name: 'kentia', _id: '1a'},
      {name: 'potus', _id: '2a'},
    ],
  });

  describe('When is invoked', () => {
    test('Then it is renders correctly', () => {
      const tree = render(
        <Provider store={store}>
          <MyPlantCard />
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
