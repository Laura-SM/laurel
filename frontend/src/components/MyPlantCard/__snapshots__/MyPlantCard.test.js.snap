// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a MyPlantCard component When invoques MyPlantCard function Then MyPlantCard component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#93B1A7",
      "borderRadius": 10,
      "height": 230,
      "justifyContent": "flex-end",
      "marginLeft": 20,
      "marginVertical": 10,
      "width": 165,
    }
  }
  testID="goToDetails"
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "height": 165,
        "position": "absolute",
        "top": 0,
        "width": 165,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginHorizontal": 20,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginBottom": 10,
        "marginHorizontal": 20,
      }
    }
  />
</View>
`;
